explore_text_columns

charlessuresh opened the pull request:
- Wrote test cases and coded iteratively
- added exception handling
- updated function specification
- updated `explore_text_columns` section of README
- tested function in `eda_playground.Rmd`

The functions `explore_text_columns` and `test-explore_text_columns` are a bit long. However, because of time limitations, I don't think I can break them up into smaller functions in time for Milestone 3. I will probably reserve that for Milestone 4

@lephanthuymai please review
@rahulkuriyedath @adibns feel free to review as well

lephanthuymai commented on tests/testthat/test-explore_text_columns.R:
```diff
@@ -1,3 +1,47 @@
+require(tidyverse)
+require(vdiffr)
+
 test_that("multiplication works", {
```
Please change the description of the test.

charlessuresh replied:
Thanks for pointing this out!
I've addressed this in this [commit](https://github.com/UBC-MDS/datascience.eda.R/pull/15/commits/b0ae900e78f9028610288ce5fe56634ae825dac8)

lephanthuymai commented on R/explore_text_columns.R:
```diff
@@ -1,19 +1,311 @@
+  }
+
+  if(length(text_cols) == 0) {
+    for (col in colnames(df %>% dplyr::select_if(is.character))){
```
I would suggest adding some comments here on how you retrieve the text columns.

charlessuresh replied:
I've added comments to the entire code now
I've addressed this is this [commit](https://github.com/UBC-MDS/datascience.eda.R/pull/15/commits/84f285d351b363b8399d26529026ad0be8e11c7a)

lephanthuymai commented on R/explore_text_columns.R:
```diff
@@ -1,19 +1,311 @@
+      if (!(col %in% colnames(df))){
+        stop(paste0(col, " passed in text_cols is not a column in the dataframe"))
+      }
+      if (!is.character(df)){
```
was this meant to be !is.character(col) instead of !is.character(df)?

charlessuresh replied:
Good spot!
I've addressed this is in this [commit](https://github.com/UBC-MDS/datascience.eda.R/pull/15/commits/3ca7a07b6426286a79a5b8b44515395ebef1f3ff)

lephanthuymai commented on R/explore_text_columns.R:
```diff
@@ -1,19 +1,311 @@
-#' explore_text_columns(df)
-#' }
-explore_text_columns <- function(df, text_cols=NULL){
+#' explore_text_columns(cars)
```
should we give another example? as there is no text column in cars

charlessuresh replied:
I've added another example using the "apocrypha" dataset from a developer package [JohnCoene/sacred](https://github.com/JohnCoene/sacred)
I've addressed this is in this [commit](https://github.com/UBC-MDS/datascience.eda.R/pull/15/commits/5d271e4a9a0db4637edfa95e1f331eb4d6a8285a)

lephanthuymai reviewed (commented):
Overall, great work, Charles!
I feel like there is a lack of comments in your code, adding some describing what will happen in each part of your function will be helpful for others to review your code (especially the TA).
Thank you for your hard work on this.
Best,
Mai

charlessuresh commented:
Hey @lephanthuymai 

Thanks for the detailed and in-depth review. I've addressed the issues pointed out via the latest commits.
Request you to review the corrections/additions made.

adibns commented:
Nothing much to say from my end. Awesome work @charlessuresh
